odl/util/utility.py

```diff
@@ -707,6 +708,15 @@ def signature_string(posargs, optargs, sep=', ', mod=''):
     return part_sep.join(parts)
 
 
+def run_from_ipython():
+    """If the process is run from IPython."""
+    try:
+        __IPYTHON__
+        return True
+    except NameError:
+        return False
```

Comment:
Not sure if better, but shorter and more to the point. In the end, your version also tries to look up the name `'__IPYTHON__` in the global (module) dictionary, and if present, returns `True`, otherwise `False`. So unless I'm missing something regarding `globals()` I think it's the same.